public/app/features/dashboard/state/runRequest.ts

```diff
@@ -226,3 +227,34 @@ export function preProcessPanelData(data: PanelData, lastResult: PanelData): Pan
     timings: { dataProcessingTime: STOPTIME - STARTTIME },
   };
 }
+
+export function preProcessAppendPanelData(
```

Comment:
This is very similar to `preProcessPanelData` and should be refactored so there isn't so much code duplication.

Reply:
Yep, will do.